test/e2e/util.go

```diff
@@ -401,10 +404,16 @@ func (b *albcBuilder) withCredSecret(name string) *albcBuilder {
 	return b
 }
 
-// withCredSecretIf adds the credentials secret only if the given condition is true.
-func (b *albcBuilder) withCredSecretIf(condition bool, name string) *albcBuilder {
+// withRoleARNIf adds given role ARN to the credentials request config if the given condition is true.
+func (b *albcBuilder) withRoleARNIf(condition bool, roleARN string) *albcBuilder {
```

Comment:
So I tried to run E2E in my STS cluster, non-ROSA, but this logic won't allow me to provide a roleARN manually, right?

I feel like we should support directly provide ENV while not in a Rosa cluster, otherwise I can't test your changes manually in my STS cluster.

Reply:
There is [a dedicated README chapter](https://github.com/openshift/aws-load-balancer-operator#running-the-end-to-end-tests-on-rosa-sts-cluster) for the manual testing on a ROSA cluster. But I believe it should work on any STS cluster too. I think I should add `ALBO_E2E_CONTROLLER_ROLE_ARN` variable in there and maybe make it more generic: change the chapter name and add new platform to`ALBO_E2E_PLATFORM` envvar.

Does it sound like what you were looking for?

Reply:
Ah okay, I didn't realize the platform was controlled by a ENV Variables, I thought it was looking at some sort of cluster config object. Yea I suppose I could say I'm "ROSA" to test, but that does feel a little funky because I'd be lying.

I think your suggestion makes sense. If I provide `ALBO_E2E_CONTROLLER_ROLE_ARN`, I feel like it should use that in ALBC creation regardless of my platform, but at the same time, I don't think I should have to set `ALBO_E2E_PLATFORM` to `ROSA`.

Reply:
> Ah okay, I didn't realize the platform was controlled by a ENV Variables, I thought it was looking at some sort of cluster config object.

I don't know any reliable way of how to detect whether your container runs on a cluster which uses STS credentials. `Infrastructure` resource doesn't have any info about it. CCO in manual mode  != STS. We could check the format of the contents of the cluster operator's secrets but that seems to be a little too much. Overall, it's a good question which needs to be digged in. I'll try to follow up on it.

I tried to address the comment in [this commit](https://github.com/openshift/aws-load-balancer-operator/pull/113/commits/6cb72b036ef93e0c8278af83ce6f08a5fe839ba7). Let me know if it's still unclear.

Reply:
Yea your approach is fine.